packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +829,385 @@ def to_json(self):
         return obj_dict
 
 
+def request_builder(
+    self, json_request, yaml, not_first, indent=4, name="", inner_model=[], no_list=True
+):
+    if inner_model and not self.inner_model: self.inner_model = inner_model
+    no_list = list_format(self, json_request, yaml, not_first, indent, name, inner_model)
+
+    not_first, indent, inner_model = dict_format(self, json_request, yaml, not_first, indent, inner_model, no_list,name)
+
+def dict_format(self, json_request, yaml, not_first, indent, inner_model, no_list,name):
+    if isinstance(json_request, dict):
+        for i in json_request:
+            if indent == 4:
+                self.add_whitespace(indent)
+                if not_first:
+                    self.add_new_line()
+                    self.add_whitespace(indent)
+                    self.add_new_line()
+                    self.add_whitespace(indent)
+                if not inner_model:
+                    self.add_comment(None, "model " + i + " {", None)
+                    self.add_new_line()
+                    not_first = True
+                    name = i
+                    inner_model.clear()
+                else:
+                    inner_model.append(Token(" ", TokenKind.Newline))
+                    inner_model.append(Token(" " * (indent * 4), TokenKind.Whitespace))
+                    inner_model.append(Token("model " + i + " {", TokenKind.Comment))
+                    inner_model.append(Token(" ", TokenKind.Newline))
+            if indent > 4 and (
+                isinstance(json_request[i], list) or isinstance(json_request[i], dict)
+            ):
+                if i == "str":
+                    if inner_model:
+                        inner_model.append(Token(" ", TokenKind.Newline))
+                        inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+                        m_type, key = get_map_type(yaml, name)
+                        m_type = format_map_type(m_type)
```

Comment:
it doesn't seem like `m_type` is used after being defined

Reply:
it's used in inner_model